src/h3/ffi.rs

```diff
@@ -121,6 +123,8 @@ pub extern fn quiche_h3_event_for_each_header(
             },
 
         h3::Event::Data { .. } => unreachable!(),
+
+        h3::Event::StreamClosed { .. } => unreachable!(),
```

Comment:
You could probably just replace this and the previous case with `_ => unreachable!()`.